core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -97,4 +111,34 @@ private ProbeResult fromSCMLink(ProbeContext context, String scm) {
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory  in the scm
+     * @param pluginName the name of the plugin
+     * @param scm        the valid scm link
+     * @return folderPath if it valid or return the scm itself
+     */
+    private String searchPomFiles(Path directory, String pluginName, String scm) {
+        MavenXpp3Reader mavenReader = new MavenXpp3Reader();
+
+        try (Stream<Path> paths = Files.find(directory, 1, (path, $) ->
```

Comment:
With a `maxDepth` of `1`, you will just search the root directory, so this is not doing what you want it to do.

Did you write proper tests to check what you wanted to do? Are they passing?